salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
Just append a commit should be fine, just give it a proper description. (e.g. Allow use of by-uuid with no vdev type)